src/Database/LSMTree/Internal/BlobRef.hs

```diff
@@ -188,17 +188,16 @@ readRawBlobRef ::
 readRawBlobRef fs RawBlobRef {rawBlobRefFile, rawBlobRefSpan} =
     BlobFile.readBlobFile fs rawBlobRefFile rawBlobRefSpan
 
-{-# SPECIALISE readBlob ::
-     HasFS IO h
-  -> StrongBlobRef IO h
-  -> IO SerialisedBlob #-}
-readBlob ::
-     (MonadThrow m, PrimMonad m)
+{-# SPECIALISE readWeakBlobRef :: HasFS IO h -> WeakBlobRef IO h -> IO SerialisedBlob #-}
+readWeakBlobRef ::
+     (MonadMask m, PrimMonad m)
   => HasFS m h
-  -> StrongBlobRef m h
+  -> WeakBlobRef m h
   -> m SerialisedBlob
-readBlob fs StrongBlobRef {strongBlobRefFile, strongBlobRefSpan} =
-    BlobFile.readBlobFile fs strongBlobRefFile strongBlobRefSpan
+readWeakBlobRef fs wref =
+    bracket (deRefWeakBlobRef wref) removeReference $
```

Comment:
This is `withWeakBlobRef`. I noticed that a later commit removes `withWeakBlobRef`, however

Reply:
Yes. This is the only place it's used. Fewer lines of code this way (no type sigs and no specialise pragmas).